Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -839,6 +839,16 @@ class BrowserViewController: UIViewController {
                     self.tabManager.addTabsForURLs(receivedURLs, zombie: false)
                 }
             }
+
+            if !receivedURLs.isEmpty || cursorCount > 0 {
+                // Because the notification service runs as a seperate process
+                // we need to make sure that our account manager picks up any persisted state
+                // the notification services persisted.
+                DispatchQueue.main.async {
+                    self.profile.rustFxA.accountManager.peek()?.resetPersistedAccount()
+                    self.profile.rustFxA.accountManager.peek()?.deviceConstellation()?.refreshState()
+                }
```

Comment:
Is this something we have to do on main thread? 🤔

Reply:
Good catch, no - I should have caught that the "open in tab" dispatch was there for a reason! removed the dispatch and ran those calls directly since the calling thread is a background thread anyways